src/appleseed/renderer/kernel/shading/shadingpoint.cpp

```diff
@@ -813,6 +813,15 @@ void ShadingPoint::compute_shading_basis() const
     const Vector3d s = normalize(cross(sn, t));
     m_shading_basis.build(sn, s, t);
 
+    if (m_primitive_type == PrimitiveCurve3)
```

Comment:
We shouldn't build an orthonormal basis only to overwrite it later in the case of curves.